src/test/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/CalculationTransactionalServiceTest.kt

```diff
@@ -157,6 +158,44 @@ class CalculationTransactionalServiceTest {
     assertEquals(bookingData.effectiveSentenceLength, calculatedReleaseDates.effectiveSentenceLength)
   }
 
+  @ParameterizedTest
+  @CsvFileSource(resources = ["/test_data/calculation-validation-examples.csv"], numLinesToSkip = 1)
+  fun `Test validation after calculations by example`(exampleType: String, exampleNumber: String, error: String?, params: String) {
+    log.info("Testing example $exampleType/$exampleNumber")
+    whenever(calculationRequestRepository.save(any())).thenReturn(CALCULATION_REQUEST)
+    whenever(serviceUserService.getUsername()).thenReturn(USERNAME)
+
+    val (booking, calculationUserInputs) = jsonTransformation.loadBooking("$exampleType/$exampleNumber")
+    val calculatedReleaseDates: CalculatedReleaseDates
+    val returnedValidationMessages: List<ValidationMessage>
+    try {
+      calculatedReleaseDates = calculationTransactionalService(params, passedInServices = listOf(ValidationService::class.java.simpleName))
+        .calculate(booking, PRELIMINARY, fakeSourceData, CALCULATION_REASON, calculationUserInputs)
+      val sentencesExtractionService = SentencesExtractionService()
+      val trancheOne = TrancheOne(sdsEarlyReleaseTrancheOneDate(params))
+      val myValidationService = getActiveValidationService(sentencesExtractionService, trancheOne)
+      returnedValidationMessages = myValidationService.validateBookingAfterCalculation(
+        calculatedReleaseDates.calculatedBooking!!,
+      )
+    } catch (e: Exception) {
+      if (!error.isNullOrEmpty()) {
+        assertEquals(error, e.javaClass.simpleName)
+        return
+      } else {
+        throw e
+      }
+    }
+    log.info(
+      "Example $exampleType/$exampleNumber outcome BookingCalculation: {}",
+      TestUtil.objectMapper().writeValueAsString(calculatedReleaseDates),
+    )
+    val bookingData = jsonTransformation.loadCalculationResult("$exampleType/$exampleNumber")
+
+    assertEquals(bookingData.dates, calculatedReleaseDates.dates)
+    assertEquals(bookingData.effectiveSentenceLength, calculatedReleaseDates.effectiveSentenceLength)
+    assertThat(returnedValidationMessages).hasSize(1) // etc
```

Comment:
I'm guessing that this didn't run ok? Else you'd be expecting to get an error message I thought your changes to get the longest possible sentences to check date of award etc

Reply:
Unless I'm not getting the test that is

Reply:
As clearly the checks completed ok 👀